offset/offset_clock.go

```diff
@@ -49,6 +49,14 @@ func (o *Clock) AfterFunc(d time.Duration, f func()) clocks.StopTimer {
 	return o.inner.AfterFunc(d, f)
 }
 
+func (o *Clock) ContextWithDeadline(ctx context.Context, t time.Time) (context.Context, context.CancelFunc) {
+	return o.inner.ContextWithDeadline(ctx, t.Add(o.offset))
+}
+
+func (o *Clock) ContextWithTimeout(ctx context.Context, d time.Duration) (context.Context, context.CancelFunc) {
+	return o.inner.ContextWithTimeout(ctx, d+o.offset)
```

Comment:
Hmm, I don't think this one needs any adjustment. (since it's already a relative duration)